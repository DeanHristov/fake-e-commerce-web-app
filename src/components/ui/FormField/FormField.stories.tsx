import type { Meta, StoryObj } from '@storybook/react';

import FormField, { IFormFieldProps } from './index';

const meta: Meta<IFormFieldProps> = {
  title: 'UI/FormField',
  component: FormField,
  parameters: {},
  argTypes: {
    onChange: {
      action: 'onChange():',
    },
  },
};

type Story = StoryObj<IFormFieldProps>;

export const Primary: Story = {
  args: {
    label: 'Username or Email',
  },
};

export default meta;
