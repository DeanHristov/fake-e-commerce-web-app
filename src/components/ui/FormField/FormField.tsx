'use client';

import { FC } from 'react';
import InputField, { IInputFieldProps } from '../InputField';

export interface IFormFieldProps extends IInputFieldProps {
  label: string;
  className?: string;
}

const FormField: FC<IFormFieldProps> = ({ label, onChange, className }) => {
  return (
    <div className={`py-2 ${className ?? ''}`}>
      <label htmlFor="email" className="block text-xs mb-1 cursor-pointer">
        {label}
      </label>
      <div className="h-12">
        <InputField id="email" onChange={onChange} />
      </div>
    </div>
  );
};

export default FormField;
