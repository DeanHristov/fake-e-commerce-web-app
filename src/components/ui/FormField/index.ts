import FormField, { IFormFieldProps } from './FormField';

export type { IFormFieldProps };
export default FormField;
