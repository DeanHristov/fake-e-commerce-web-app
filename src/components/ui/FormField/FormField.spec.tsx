import { render } from '@testing-library/react';
import FormField from './index';

describe('UI/FormField <FormField {...} />', () => {
  it('Should render the badge with a counter', () => {
    const { container } = render(
      <FormField label="Username or an email" onChange={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
