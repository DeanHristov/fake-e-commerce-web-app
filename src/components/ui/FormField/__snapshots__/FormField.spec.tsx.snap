// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/FormField <FormField {...} /> Should render the badge with a counter 1`] = `
<div>
  <div
    class="py-2"
  >
    <label
      class="block text-xs mb-1 cursor-pointer"
      for="email"
    >
      Username or an email
    </label>
    <div
      class="h-12"
    >
      <div
        class="relative w-full rounded-lg h-full border border-gray-500"
      >
        <input
          class="h-full px-6 w-full rounded-lg bg-white text-slate-700 ease-in duration-200 focus:bg-white"
          id="email"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
